### ISSUE TITLE
Coverage not generated when --coverage not used

### ISSUE THREAD

**Issue by luispabon** [infection/infection#299]
| Question    | Answer
| ------------| ---------------
| Infection version |0.8.1
| Test Framework version | PHPUnit 7.1.1
| PHP version | 7.2.4
| Platform    | Ubuntu
| Github Repo | -

I'm currently wiring up to infection 0.8.1 up from 0.7 to take advantage of the new mutators and the pre-existing coverage functionality.

I've run into behaviour I wasn't expecting - coverage is not automatically generated when I'm not specifying pre-existing coverage. The behaviour I was expecting is to only rely on externally generated coverage reports only when `--coverage` is used, or default to pre 0.8 behaviour when it isn't.

```
$ bin/infection --threads=8 -s       
You are running Infection with xdebug enabled.
    ____      ____          __  _
   /  _/___  / __/__  _____/ /_(_)___  ____ 
   / // __ \/ /_/ _ \/ ___/ __/ / __ \/ __ \
 _/ // / / / __/  __/ /__/ /_/ / /_/ / / / /
/___/_/ /_/_/  \___/\___/\__/_/\____/_/ /_/
 
Running initial test suite...

PHPUnit version: unknown

    1 [============================] < 1 sec

Generate mutants...

Processing source code files: 149/149
Creating mutated files and processes:    0/1386
In CoverageDoesNotExistException.php line 18:
                                                                                                                                                                                         
  Code Coverage does not exist. File /tmp/infection/coverage-xml/index.xml is not found. Check phpunit version Infection was run with and generated config files inside /tmp/infection.  
                                                                                                                                                                                         
```

<details>
 <summary>phpunit.xml</summary>
 
 ```xml
<?xml version="1.0" encoding="UTF-8"?>

<!-- https://phpunit.de/manual/current/en/appendixes.configuration.html -->
<phpunit xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
         xsi:noNamespaceSchemaLocation="http://schema.phpunit.de/6.1/phpunit.xsd"
         backupGlobals="false"
         colors="true"
         bootstrap="vendor/autoload.php"
>
    <php>
        <ini name="error_reporting" value="-1" />
        <env name="KERNEL_CLASS" value="App\Kernel" />
        <env name="APP_ENV" value="test" />
        <env name="APP_DEBUG" value="1" />
        <!-- define your env variables for the test env here -->

        <!-- ###+ nelmio/cors-bundle ### -->
        <env name="CORS_ALLOW_ORIGIN" value="^https?://localhost:?[0-9]*$"/>
        <!-- ###- nelmio/cors-bundle ### -->
    </php>

    <filter>
        <whitelist>
            <directory>src</directory>
        </whitelist>
    </filter>

    <testsuites>
        <testsuite name="unit">
            <directory>tests/App/Unit/</directory>
            <directory>tests/EDITED/Unit/</directory>
        </testsuite>
        <testsuite name="functional">
            <directory>tests/App/Functional/</directory>
            <directory>tests/EDITED/Functional/</directory>
        </testsuite>
        <testsuite name="framework">
            <directory>tests/App/</directory>
        </testsuite>
        <testsuite name="non-framework">
            <directory>tests/EDITED/</directory>
        </testsuite>
    </testsuites>

    <listeners>
        <listener class="NunoMaduro\Collision\Adapters\Phpunit\Listener" />
    </listeners>

    <!-- Reports -->
    <logging>
        <log type="testdox-html" target="reports/phpunit/testdox.html"/>
        <log type="coverage-html" target="reports/phpunit/coverage" lowUpperBound="85" highLowerBound="90" />
        <log type="coverage-clover" target="reports/phpunit/coverage.xml"/>
        <log type="coverage-crap4j" target="reports/phpunit/crap4j.xml"/>
        <log type="junit" target="reports/phpunit/junit.xml"/>
    </logging>
</phpunit>

 ```
</details>
<details>
 <summary>infection.json</summary>
 
 ```json
{
  "source":  {
    "directories": [
      "src"
    ],
    "excludes": [
      "App/Kernel.php",
      "App/Listener/Response/ValidatorListener.php",
      "App/Listener/Response/FrameworkResponseListener.php"
    ]
  },
  "timeout": 10,
  "logs":    {
    "text": "reports/infection-log.txt"
  },
  "phpUnit": {
    "configDir":  ".",
    "customPath": "vendor/bin/phpunit"
  }
}
 ```
</details>

**Comment by luispabon** [comment]
Hmm I wonder if this is a red herring? My PHPUnit executable as found in my path is a bash script that runs `vendor/bin/phpunit` if available and falls back to a globally installed `phpunit.phar` if it isn't.

I can see infection is having trouble with this as it's wrapping `phpunit` calls with `php` which obviously won't work on this set up. Furthermore, `infection.json:phUnit.customPath` seems to be completely ignored and indeed is never used by infection in 0.8.x (I can see this is fixed in dev-master).

**Comment by theofidry** [comment]
I'm a bit confused, is the issue fixed on dev-master?

**Comment by luispabon** [comment]
Not exactly. The root problem is infection assuming `$PATH/phpunit` is a php script/phar and not something else (like a bash script) and it's compounded by the fact `infection.json:phUnit.customPath` doesn't work in 0.8.x. Since this is the current "stable" release it leaves me unable to upgrade until 0.9.x comes out.

Another side effect of this is the fact that if I do provide with a pre-generated coverage report, infection doesn't notice phpunit isn't actually executing at all when it runs the tests on each mutated codebase - it gives back MSI of roughly 100%.

Long story short, the real issues are:
  * `infection.json:phUnit.customPath` doesn't work in 0.8.x
  * Infection does not check phpunit is executable with php.

Will open issues on these and close this one.
